File: HorseRacing.Web/src/interfaces/api/contracts/model/game/requests/create-game/create-game-request.ts
import type { BaseRequest } from "../../../../base/requests/base-request";
import type { CreateGameRequestDto } from './create-game-request-dto';

export class CreateGameRequest implements BaseRequest<CreateGameRequestDto> {
    Data: CreateGameRequestDto;

    constructor(data?: CreateGameRequestDto) {
        this.Data = data || { UserId: '', Name: '', BetAmount: 0, BetSuit: 0 };
    }
}

File: HorseRacing.Contracts/Models/Game/Requests/CreateGame/CreateGameRequest.cs
using HorseRacing.Contracts.Base.Requests;

namespace HorseRacing.Contracts.Models.Game.Requests.CreateGame
{
    public class CreateGameRequest : BaseRequest<CreateGameRequestDto>
    {
        public CreateGameRequest() : base() { }
        public CreateGameRequest(CreateGameRequestDto data) : base(data) { }
    }
}
